frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx

```diff
@@ -20,22 +26,76 @@ function FormattedParameterValue({
   value,
   placeholder,
 }: FormattedParameterValueProps) {
+  const hasMultipleValues = Array.isArray(value) && value.length !== 1;
+
+  const { data, isLoading, isError } = useGetParameterValuesQuery(
```

Comment:
I can't find this getting called. Could you give an example where it's needed? This looks wrong/suspicious for multiple reasons:
1. The main app's endpoint is used unconditionally while in public & embedded questions & dashboards there are specialized endpoints
2. Casting `field.id` to number means it won't work for anything based on a SQL query/model

Reply:
Overall this type of fetching _looks_ like an API design mistake and should be probably avoided; but unclear for now until there are answers for the question above.

Reply:
Yes, it's getting called incorrectly in public dashboards 
<img width="1506" alt="Screenshot 2024-07-01 at 09 31 45" src="https://github.com/metabase/metabase/assets/8542534/e1e060c9-2a82-4992-a090-ab45cece28ef">

Reply:
Hmm, I don't see how we can fetch the label for the field then.

FWIW, we already fetch the values when actually editing the filter to get the list of values to show in the dropdown when the search type is dropdown list.
On my machine that endpoint does not return a 401 error.

What would a more correct approach be here then? Without the possibility of loading labels I don't think this feature is possible.

Are we explicitly trying to avoid loading values from dashboards? Or am I just using the wrong endpoint here (if so what is the correct endpoint to use)?

Reply:
@romeovs could you explain why this is even needed? It's skipped when there is 'custom list', so for your change it's never called

Reply:
TBH, I added it as a bit of defensive programming, but it might not be necessary.



The `FormattedParameterValue` component looks at `props.parameter.values_source_config` to find the label for a value, but the `parameter` prop is optional, and `parameter.values_source_config` is also marked as optional. So I figured we need a way to fetch the data if `values_source_config` has not been passed through props.

Maybe we can leave this out and count on the fact that in cases the `parameter.values_source_config` values is not set, the endpoint won't work either?

Reply:
could you revert all changes to this component? it seems they are not used at all

Reply:
if they are needed, please tell me the steps which would lead to this call being used

Reply:
They are needed to render the custom label on the filter value picker component.

1. Set up a parameter with custom labels
2. Pick a value with a custom label
3. Ensure the custom label is shown on the parameter widget

Otherwise it might look like:
<img width="348" alt="Screenshot 2024-07-01 at 17 33 42" src="https://github.com/metabase/metabase/assets/1250185/858302a5-9ae8-41d5-9763-b017107d7791">

Reply:
This is 100% not the case - `skip` would be `true`.